File: load.pl
#! /usr/bin/env perl

use v5.22;
use warnings;
use re '/aa';
use Term::ANSIColor qw/color :constants/;

# CPU info
open my $FH, '<', '/proc/cpuinfo' or die RED.$!.RESET, "\n";

# load average
chomp (my @load = split ' ', `cat /proc/loadavg`);
my ($one, $five, $fifteen) = @load;

my $msg = 'increasing' if $one > $five or $one > $fifteen;

# get cores count
my $procs;
while (<$FH>) {
   chomp;
   ++$procs if /^processor\h*:\h*\d/;
}

map { $_ = ($procs > $_ ? GREEN : RED).$_.RESET } @load[0..2];

say '   1,    5,   15 : '.ITALIC.'minutes'.RESET;
say join (', ', @load[0..2]),
    ' : '.BOLD.$procs.RESET, ' cores ', ITALIC.'load average',
    ($msg ? " ($msg)" : '').RESET;

File: load.py
#! /usr/bin/env python3

'''Load average with clear time slices and trend'''

from os import cpu_count

esc = '\033['
GREEN, RED, BOLD, ITALIC, RESET = [f'{esc}{code}m' for code in (32, 31, 1, 3, 0)]

# load average
with open('/proc/loadavg') as file:
   load = file.read().split() [:3]

one, five, fifteen = map(float, load)

trend = ''
if one > five or one > fifteen:
   trend = '(increasing)'

# get cores count
count = cpu_count()
cores = 'cores' if count > 1 else 'core'

# Output
load = [ (GREEN if count > float(ld) else RED) + ld + RESET for ld in load ]

print('   1,    5,   15 :', ITALIC + 'minutes' + RESET)
print(', '.join(load), ':', BOLD + str(count) + RESET, cores,
      ITALIC + 'load average', trend + RESET)
